internal/pkg/config/specifiers.go

```diff
@@ -0,0 +1,78 @@
+package config
+
+import (
+	"github.com/google/go-flow-levee/internal/pkg/config/regexp"
+)
+
+// This type marks intended future work
+type NotImplemented = interface{}
+
+// ConfigV2 is a more generic config.
+type ConfigV2 struct {
+	ApiVersion string `yaml:"apiVersion"`
+	Kind       string
+
+	Metadata metaSpec
+
+	Source    []specifier
+	Sink      []specifier
+	Sanitizer []specifier
+	Allowlist []specifier
+}
+
+type metaSpec struct {
+	ValidateConfig NotImplemented `yaml:"validateConfig"`
+	Scope          NotImplemented
+}
+
+type valueSpec struct {
+	Id            string           // Identifier for this spec
+	TypeSpec      `yaml:",inline"` // Match source type by path and type name
+	FieldSpec     `yaml:",inline"` // Match field of above type by name or tags
+	Scope         NotImplemented   // Match by local/param/global
+	IsReference   NotImplemented   // Match by pointer/slice/map
+	MatchConstant NotImplemented   // Match explicit value, e.g. "PASSWORD"
```

Comment:
Could.  I suspect that would massage some implementation details.  For instance, `ssa.Member`s doesn't implement `ssa.Node`, so the interface that will be required to check a source by type is likely going to be different than that used to check an `ssa.Value`.

Reply:
Counter-point, I'm trying to keep the config a bit implementation-agnostic and accessible.  If we split out `type`, `field`, `value`, and `call`, it begins to feel a bit precious.

```yaml
source:
- type:   # identify all fizz.buzz as sources
    package: fizz
    struct: buzz
- field:   # Type foo.bar contains field baz, which is a source
    package:  foo
    struct: bar
    fieldName: baz
- field:    # Any type containing a specific tag should be marked a source
    fieldTag:
      - qux: thud
- call:  # The string returned by os.getEnv("CFSSL_CA_PK_PASSWORD") is sensitive
    package: os
    function: getEnv
    arguments:
    - value:
        position: 0
        const: CFSSL_CA_PK_PASSWORD
```

That feels a bit redundant between the `field` and `type` spec.

Their inclusion in this draft implementation was just an attempt to extract shared components, not necessarily to expose that as part of the configuration api.

Reply:
SG. I wasn't really thinking about the implementation TBH.

To me, being able to know up front the *kind* of value that is being defined makes it easier to understand *what* is being defined. Reading the example in your comment, it is easy for me to say "this is a *type*", "this is a *field*", and then when I read the `key: value` pairs I can fill in the details of what specifically is being defined. Whereas if I only have `value`, I have to look at the `key: value` pairs to deduce what kind of thing is being defined. In the case of `field` vs `type`, until I get to `fieldName`, I have no idea what kind of thing is being defined.

For structs we may want to have a list of sensitive field names, in case there are many sensitive fields.

I feel like this discussion is getting beyond the current scope of the PR, we can defer if you want.